tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py

```diff
@@ -786,6 +900,100 @@ def test_hetero_graph_on_shared_memory(
     assert metadata.edge_type_to_id == graph2.metadata.edge_type_to_id
 
 
+def process_csc_sampling_graph_with_queue(graph, graph_queue, flag_queue):
+    # Put the graph into the queue.
+    graph_queue.put(graph)
+    # Wait for the main process to finish.
+    flag_queue.get()
+
+
+def check_spawned_tensors_on_the_same_shared_memory(t1, t2):
+    """Check if two tensors are on the same shared memory.
+
+    This function copies a random tensor value to `t1` and checks whether `t2`
+    holds the same random value and checks whether t2 is a distinct tensor
+    object from `t1`. Their equality confirms that they are separate tensors
+    that rely on the shared memory for their tensor value.
+    """
+    old_t1 = t1.clone()
+    v = torch.randint_like(t1, 100)
+    t1[:] = v
+    assert torch.equal(t1, t2)
+    t1[:] = old_t1
+
+
+@unittest.skipIf(
+    F._default_context_str == "gpu",
+    reason="Graph is CPU only at present.",
+)
+def test_multiprocessing_with_shared_memory():
+    """Test if two CSCSamplingGraphs are on the same shared memory after
+    spawning.
+
+    For now this code only works when the sharing strategy of
+    torch.multiprocessing is set to `file_system` at the beginning.
+    The cause is still yet to be found.
+    """
+
+    num_nodes = 5
+    num_edges = 10
+    num_ntypes = 2
+    num_etypes = 3
+    (
+        csc_indptr,
+        indices,
+        node_type_offset,
+        type_per_edge,
+        metadata,
+    ) = gbt.random_hetero_graph(num_nodes, num_edges, num_ntypes, num_etypes)
+
+    csc_indptr.share_memory_()
+    indices.share_memory_()
+    node_type_offset.share_memory_()
+    type_per_edge.share_memory_()
+
+    graph = gb.from_csc(
+        csc_indptr,
+        indices,
+        node_type_offset,
+        type_per_edge,
+        None,
+        metadata,
+    )
+
+    ctx = mp.get_context("spawn")  # Use spawn method.
+
+    graph_queue = ctx.Queue()  # Used for sending graph.
+    flag_queue = ctx.Queue()  # Used for sending finish signal.
+
+    p = ctx.Process(
+        target=process_csc_sampling_graph_with_queue,
+        args=(graph, graph_queue, flag_queue),
+    )
+    p.start()
+    try:
+        graph2 = graph_queue.get()  # Get the graph from the other process.
```

Comment:
If you get the graph from a subprocess through queue, the graph is on the same shared memory as your test graph. Then the testing is meaningless. I'd suggest you to randomly assign values to the tensors of the graph in the subprocess and test whether the test graph in the main process has the same values.